### PR TITLE
Fix narrow issue in `MultiNestedTensor`

### PR DESCRIPTION
**Pull request by rusty1s**

### CODE REVIEW

File: torch_frame/data/multi_tensor.py [hunk 1]
```diff
@@ -143,9 +143,8 @@ def _normalize_index(
         if isinstance(index, int):
             if index < 0:
                 index = index + max_entries
-            if is_slice_end and index < 0 or index > max_entries:
-                raise IndexError(f"{idx_name} index out of bounds!")
-            elif (not is_slice_end) and (index < 0 or index >= max_entries):
+            index = min(index, max_entries)
+            if index < 0:
```
Comment (weihua916) [inline torch_frame/data/multi_tensor.py:147]: Is it normal to clamp in this way? why are we removing `is_slice_end`?

**Comment by weihua916** [comment]
See here: https://github.com/pyg-team/pytorch-frame/blob/master/torch_frame/data/multi_tensor.py#L257-L261

Below, the slice end should allow 100, but normal indexing should not.
```python
>>> import numpy as np
>>> np.zeros(100)[:100]
array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])
>>> np.zeros(100)[100]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: index 100 is out of bounds for axis 0 with size 100
```

File: torch_frame/data/multi_tensor.py (hunk 1 above)
Comment (rusty1s) [reply torch_frame/data/multi_tensor.py:147]: I see. I only checked the narrow code path, which is broken due to these checks (see the test).

**Comment by rusty1s** [comment]
@weihua916 PTAL.

**Review by akihironitta — APPROVED** [review]

**Comment by rusty1s** [comment]
Merging this since I am blocked. @weihua916 I can add any feedback as a follow-up.

**Review by weihua916 — COMMENTED** [review]
Looks good! thanks.

**Pull request merged**